app/models/mixins/supports_feature_mixin.rb

```diff
@@ -0,0 +1,56 @@
+module SupportsFeatureMixin
+  extend ActiveSupport::Concern
+
+  class_methods do
+    def supports(feature, is_supported = true, reason = "", &block)
+      method_name = "supports_#{feature}?"
+
+      self.class.send(:define_method, method_name) do
+        result = if is_supported
+                   true
+                 else
+                   reason
+                 end
+        if result === true
+          true
+        else
+          false
+        end
```

Comment:
Feels like this class method definition could be reduced to:

```ruby
self.class.send(:define_method, method_name) do
  is_supported
end
```

* `result` is either `true`, or `reason`
* `reason` is never `=== true` (unless someone passes `reason = true` in the arguments, which seems weird)
* the last if statement can be reduced to `return result === true`
* `result === true` only if `is_supported == true`
* so, `return result === true` is the same as `return is_supported`

Is that right?